packages/components/pie-switch/test/component/pie-switch.spec.ts

```diff
@@ -1,194 +1,196 @@
-import { test, expect } from '@sand4rt/experimental-ct-web';
-import { PieSwitch } from '../../src/index.ts';
-import {
-    type SwitchProps,
-    labelPlacements,
-    ON_SWITCH_CHANGED_EVENT,
-} from '../../src/defs.ts';
-
-const componentSelector = '[data-test-id="switch-component"]';
-const inputSelector = '[data-test-id="switch-input"]';
-const switchLabelSelector = (placement: SwitchProps['labelPlacement'] = 'leading') => `[data-test-id="switch-label-${placement}"]`;
+import { test, expect } from '@playwright/test';
+import { BasePage } from '@justeattakeaway/pie-webc-testing/src/helpers/page-object/base-page.ts';
+import { labelPlacements, type SwitchProps } from '../../src/defs';
+import { pieSwitch } from '../helpers/page-objects/selectors';
 
 test.describe('Component: `Pie switch`', () => {
-    test.beforeEach(async ({ mount }) => {
-        const component = await mount(PieSwitch);
-        await component.unmount();
-    });
-
-    test('should be visible', async ({ mount, page }) => {
+    test('should be visible', async ({ page }) => {
         // Arrange
-        await mount(PieSwitch);
+        const switchPage = new BasePage(page, 'switch');
+        await switchPage.load();
 
         // Act
-        const pieSwitch = page.locator(componentSelector);
+        const switchComponent = page.getByTestId(pieSwitch.selectors.container.dataTestId);
 
         // Assert
-        await expect(pieSwitch).toBeVisible();
+        await expect(switchComponent).toBeVisible();
     });
```

Comment:
I don't think this kind of test is particularly valuable as if it isn't visible in any meaningful way, every test will fail as the component failed to render